src/core/components/pin.js

```diff
@@ -209,8 +230,10 @@ module.exports = (self) => {
             }
 
             // make sure we have the object
-            dag.get(new CID(multihash), { preload: options.preload }, (err) => {
-              if (err) { return cb(err) }
+            dag.get(cid, { preload: options.preload }, (err) => {
```

Comment:
Would we ever want to preload here?  I was pinning a few 10s of 1000s of nodes during an add operation and eventually the preload servers start returning `429`s.

Also during an add operation we preload, then pin but this means pinning can preload too?